tools/python/onnx_test_data_utils.py

```diff
@@ -0,0 +1,166 @@
+import argparse
+import glob
+import os
+import sys
+
+import numpy as np
+
+from onnx import numpy_helper
+from onnx import mapping
+from onnx import TensorProto
+
+
+def dump_pb_file(filename):
+    """Dump the data from a pb file containing a Tensor."""
+
+    tensor = TensorProto()
+    with open(filename, 'rb') as f:
+        tensor.ParseFromString(f.read())
+
+    print("Name: {}".format(tensor.name))
+
+    np_array = numpy_helper.to_array(tensor)
+    print("Shape: {}".format(np_array.shape))
+    print(np_array)
+
+
+def dump_pb(dir_or_filename):
+    """Read in a protobuf file or all .pb files in a directory, convert to numpy, and dump the data."""
+
+    if os.path.isdir(dir_or_filename):
+        for f in glob.glob(os.path.join(dir_or_filename, '*.pb')):
+            print(f)
+            dump_pb_file(f)
+    else:
+        dump_pb_file(dir_or_filename)
+
+
+def numpy_to_pb(name, np_data, out_filename):
+    """Convert numpy data to a protobuf file."""
+
+    tensor = numpy_helper.from_array(np_data, name)
+    with open(out_filename, 'wb') as f:
+        f.write(tensor.SerializeToString())
+
+
+def image_to_numpy(filename, shape, channels_last, add_batch_dim):
+    """Convert an image file into a numpy array."""
+
+    import PIL.Image  # from 'Pillow' package
+
+    img = PIL.Image.open(filename)
+    if shape:
+        img = img.resize(shape, PIL.Image.ANTIALIAS)
+    img_as_np = np.array(img).astype(np.float32)
+    if not channels_last:
+        # HWC to CHW
+        img_as_np = np.transpose(img_as_np, (2, 0, 1))
+
+    if add_batch_dim:
+        # to NCHW or NHWC
+        img_as_np = np.expand_dims(img_as_np, axis=0)
+
+    return img_as_np
+
+
+def create_random_data(shape, type, minvalue, maxvalue, seed):
+    nptype = np.dtype(type)
+    np.random.seed(seed)
+    return ((maxvalue - minvalue) * np.random.sample(shape) + minvalue).astype(nptype)
+
+
+def update_name_in_pb(filename, name):
+    """Update the name of the tensor in the pb file."""
+
+    tensor = TensorProto()
+    with open(filename, 'rb') as f_in:
+        tensor.ParseFromString(f_in.read())
+        tensor.name = name
+
+    with open(filename, 'wb') as f_out:
+        f_out.write(tensor.SerializeToString())
+
+
+def get_arg_parser():
+    parser = argparse.ArgumentParser(
+        description=
+        """
+        Utilities for working with the input/output protobuf files used by onnx_test_runner.
+        
+        dump_pb: Dumps the tensor data for an individual protobuf pb file, or all pb files in a directory.
+        numpy_to_pb: Write a numpy array to a protobuf file as a tensor.
+        image_to_pb: Convert data from an image file into a protobuf file as a tensor.
+        random_to_pb: Write random data into a protobuf file as a tensor.
```

Comment:
It is better to add some example commands for each task, since only a portion of options are useful for one task. 

Another solution is to use parser.add_argument_group to split arguments into groups (like min_value, max_value and seed can grouped into random related).